TESTING.md

```diff
@@ -14,13 +14,17 @@ Overview
 - [Misc](#misc)
 
 # General information
-OpenSearch Dashboards uses [Jest](https://jestjs.io/) for unit and integration tests, [Selenium](https://www.selenium.dev/) for functional tests, and [Cypress](https://www.cypress.io/) for backwards compatibility tests.
+OpenSearch Dashboards uses [Jest](https://jestjs.io/) for unit and integration tests, [Cypress](https://www.cypress.io/) for backwards compatibility tests and functional tests and [Selenium](https://www.selenium.dev/) for some legacy functional tests (Tests should no longer be written in Selenium).
 
 In general, we recommend four tiers of tests:
 * Unit tests: Unit tests: small and modular tests that utilize mocks for external dependencies.
 * Integration tests: higher-level tests that verify interactions between systems (eg. HTTP APIs, OpenSearch API calls, calling other plugin).
 * End-to-end tests (e2e): functional tests that verify behavior in a web browser.
-* Backwards Compatibility tests: cypress tests that verify any changes are backwards compatible with previous versions.
+* Backwards Compatibility tests: tests that verify any changes are backwards compatible with previous versions.
+
+> Contributors submitting pull requests (PRs) to the codebase are required to ensure that their code changes include appropriate testing coverage. This includes, but is not limited to, unit tests, integration tests, functional tests, and backwards compatibility tests where applicable.
```

Comment:
As far as I know, only unit test will contribute to test coverage. While jest_integration tests, functional tests won't. Do we have any convention that when should we add jest_integration tests and functional tests? Or I suppose some PRs which should contain this sort of tests may only include unit tests in its changes.

Reply:
@AMoo-Miki is working on bringing functional test coverage to OSD. For now this is a notice that maintainers can point to when a change does not have sufficient coverage to merge

Reply:
It is great to see some thinking on testing.

Could we clarify code coverage and test coverage. 

Let's be sure had clear actionable  plan and developer guideline so contributors could follow.

We also need a plan for those feature,component,code does't have enough coverage what will be practical plan